saleor/warehouse/migrations/0016_stock_quantity_allocated.py

```diff
@@ -8,17 +8,28 @@
 def assign_quantity_allocated_to_stocks(apps, schema_editor):
     Stock = apps.get_model("warehouse", "Stock")
 
-    Stock.objects.update(
-        quantity_allocated=Subquery(
-            Stock.objects.filter(id=OuterRef("id"))
-            .annotate(
-                allocations_allocated=Coalesce(
-                    Sum("allocations__quantity_allocated"), 0
+    batch_size = 1000
+    last_id = 0
+    while True:
+        stock_ids = list(
+            Stock.objects.filter(id__gt=last_id)
+            .order_by("id")
+            .values_list("id", flat=True)[:batch_size]
+        )
+        if not stock_ids:
+            break
+        last_id = stock_ids[-1]
+        Stock.objects.filter(id__in=stock_ids).update(
+            quantity_allocated=Subquery(
+                Stock.objects.filter(id=OuterRef("id"))
+                .annotate(
+                    allocations_allocated=Coalesce(
+                        Sum("allocations__quantity_allocated"), 0
+                    )
                 )
+                .values("allocations_allocated")[:1]
```

Comment:
IMO, we should separate part for getting batches to another function, as it's hard to ready right now.